test/test_async.py

```diff
@@ -144,7 +144,18 @@ def handle_all_unhandled_exceptions(type, value, traceback):
             tickers = await exchange.fetch_tickers([symbol])
             dump(green(exchange.id), 'fetched', green(len(list(tickers.keys()))), 'tickers')
     else:
-        dump(yellow(exchange.id), 'fetching all tickers at once not supported')
+            dump(green(exchange.id), 'fetching all tickers by simultaneous multiple concurrent requests')
+            # Some exchanges not all the symbols can fetch tickers for
+            symbols_to_load = [symbol for symbol in exchange.symbols if not '.d' in symbol]
+            if exchange.id == 'bitmex':
+                symbols_to_load = ['BTC/USD', 'B_BLOCKSZ17', 'DASHZ17', 'ETC7D', 'ETHZ17', 'LTCZ17', 'XBJZ17', 'XBTZ17', 'XMRZ17', 'XRPZ17', 'XTZZ17', 'ZECZ17']
+            elif exchange.id == 'bl3p':
+                symbols_to_load = ['BTC/EUR']
+            elif exchange.id == 'virwox':
+                symbols_to_load = [symbol for symbol in symbols_to_load if symbol != 'CHF/SLL']
+            input_coroutines = [exchange.fetchTicker(symbol) for symbol in symbols_to_load]
+            tickers = await asyncio.gather(*input_coroutines)
```

Comment:
Even though sending 16 requests at once to fetch tickers can work, it will most likely result in a ban from Bittrex or Kraken, if you try doing the same with private requests and with L2 orderbooks...

Reply:
Interesting, it sends 16 requests, then adds the possibility for a new request at slightly less than the rate limit, to make sure over a fixed time period the specified rate limit is met for the exchange. Looking at Bittrex API documentation, I can't readily see any information about their API rate limit ? (https://bittrex.com/home/api) Multiple tests to their api using this seems to work, however long polling of continuous order books is yet to be tested (I plan to add a test to test loading all order books for all exchanges). Yes I agree with Kraken in its current implementation, can make it work for Kraken by specifically adding a lower starting 'rateLimitMaxTokens' : 8, and 'rateLimitTokens' : 8.

Reply:
> it sends 16 requests, then adds the possibility for a new request at slightly less than the rate limit, to make sure over a fixed time period the specified rate limit is met for the exchange. 

Yep, and if those 16 requests are private requests (fetching balance, placing and canceling orders, fetching orders/trades history), those will fail most of the time, because practically none of the exchanges tolerate that rate.

> Looking at Bittrex API documentation, I can't readily see any information about their API rate limit ?

Yes, they don't have it documented, but it's been practically reverse-engineered to be approx 1 request per second. Liqui will also complain upon querying at rates higher than one or two requests per second. And it takes Kraken approx 10 (!) seconds to process an order, and it has been so for past few weeks, so, i guess sending 16 of them at once is not our practical option ) I am still adding some minor edits here and there to complete the merge, but most of it is already there ;)

Reply:
Very good, agreed, perhaps we should limit the throttle for asyncio to only public calls ? And keep the old rate limit delay structure for private API calls ? How do we solve this ?

Reply:
I would say, we are fine it's just that I made that functionality optional, so users may or may not enable it at their discretion depending on their use case. So if they design a poller application for alerts or something, they can go ahead and turn it full on. And if they want to do trading, then they can leave it off until we come up with a generic solution.

Reply:
Nice very good